hathor/p2p/sync_v2/manager.py

```diff
@@ -0,0 +1,1207 @@
+        assert reactor is not None
+        self.reactor: Reactor = reactor
+        self._is_streaming: bool = False
+
+        # Create logger with context
+        self.log = logger.new(peer=self.protocol.get_short_peer_id())
+
+        # Extra
+        self._blk_size = 0
+        self._blk_end_hash = settings.GENESIS_BLOCK_HASH
+        self._blk_max_quantity = 0
+
+        # indicates whether we're receiving a stream from the peer
+        self.receiving_stream = False
+
+        # highest block where we are synced
+        self.synced_height = 0
+
+        # highest block peer has
+        self.peer_height = 0
+
+        # Latest deferred waiting for a reply.
+        self.deferred_by_key: dict[str, Deferred] = {}
+
+        # When syncing blocks we start streaming with all peers
+        # so the moment I get some repeated blocks, I stop the download
+        # because it's probably a streaming that I've just received
+        self.max_repeated_blocks = 10
+
+        # Streaming objects
+        self.blockchain_streaming: Optional[BlockchainStreaming] = None
+        self.transactions_streaming: Optional[TransactionsStreaming] = None
+
+        # Whether the peers are synced, i.e. our best height and best block are the same
+        self._synced = False
+
+        # Indicate whether the sync manager has been started.
+        self._started: bool = False
+
+        # Saves the last received block from the block streaming # this is useful to be used when running the sync of
+        # transactions in the case when I am downloading a side chain. Starts at the genesis, which is common to all
+        # peers on the network
+        self._last_received_block: Optional[Block] = None
+
+        # Saves if I am in the middle of a mempool sync
+        # we don't execute any sync while in the middle of it
+        self.mempool_manager = SyncMempoolManager(self)
+        self._receiving_tips: Optional[list[bytes]] = None
+
+        # Cache for get_tx calls
+        self._get_tx_cache: OrderedDict[bytes, BaseTransaction] = OrderedDict()
+        self._get_tx_cache_maxsize = 1000
+
+        # Looping call of the main method
+        self._lc_run = LoopingCall(self.run_sync)
+        self._lc_run.clock = self.reactor
+        self._is_running = False
+
+        # Whether we propagate transactions or not
+        self._is_relaying = False
+
+        # This stores the final height that we expect the last "get blocks" stream to end on
+        self._blk_end_height: Optional[int] = None
+
+        # Whether to sync with this peer
+        self._is_enabled: bool = False
+
+    def get_status(self) -> dict[str, Any]:
+        """ Return the status of the sync.
+        """
+        res = {
+            'is_enabled': self.is_sync_enabled(),
+            'peer_height': self.peer_height,
+            'synced_height': self.synced_height,
+            'synced': self._synced,
+            'state': self.state.value,
+        }
+        return res
```

Comment:
This is part of the sync interface. At first, we used `dict[str, Any]` to allow each sync to decide what's the best way to report its status. We might use pydantic here but I can't see much value since the caller just add these pieces of data to another JSON to return in an API.

Reply:
Got it. I guess it's fine for now, specially since the caller only uses it for json, but I think even the API return type should be properly typed at some point

Reply:
Created issue: https://github.com/HathorNetwork/hathor-core/issues/727